### PR TITLE
Add OPEN_PAGES_MENU event

### LINKED ISSUE #1077: Analytics event: `OPEN_PAGES_MENU`
## Description

The goal is to 

- add the type for event `OPEN_PAGES_MENU` to the `Events` type.
  https://github.com/WordPress/openverse/blob/839418fabb0972ec3c3ccf48c72fd64cbba03cb9/frontend/src/types/analytics.ts#L12

- fire this event from the appropriate location in the codebase using the `useAnalytics` composable.
  https://github.com/WordPress/openverse/blob/839418fabb0972ec3c3ccf48c72fd64cbba03cb9/frontend/src/composables/use-analytics.ts#L11

## Reference

https://github.com/WordPress/openverse/blob/839418fabb0972ec3c3ccf48c72fd64cbba03cb9/rfcs/analytics/20221006-implementation_plan_frontend.md?plain=1#L286

## Additional information

- [RFC](https://github.com/WordPress/openverse/blob/main/rfcs/analytics/20221006-implementation_plan_frontend.md#events)
- [Example usage in `VHomeGallery.vue`](https://github.com/WordPress/openverse/blob/839418fabb0972ec3c3ccf48c72fd64cbba03cb9/frontend/src/components/VHomeGallery/VHomeGallery.vue#L128-L134)

### PR DESCRIPTION
**Pull request by zackkrida**
## Fixes

Fixes #1077 by @dhruvkb 

## Description

Adds an analytics event for the opening of the pages menu. This is a quite simple one to implement.

## Testing Instructions

Run the frontend locally, reduce the viewport width to display the pages menu toggle, and activate the toggle. Observe the event in the browser console logs.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [x] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by obulat** [comment]
@zackkrida, because of this issue I learned that some languages use NBSP (non-breaking space) between words and punctuation marks (more context in GlotPress: https://github.com/GlotPress/GlotPress/issues/801). Since `json` locale files are gitignored anyways, it seems appropriate to ignore them from linting, too.

File: .eslintignore [hunk 1]
```diff
@@ -7,3 +7,5 @@ frontend/storybook-static
 !frontend/.storybook
 
 !.pnpmfile.cjs
+
+frontend/src/locales
```
Comment (obulat) [inline .eslintignore:11]:
```suggestion
frontend/src/locales/*.json
```
There are some js files in the locales folder that _should_ be linted.

File: frontend/src/components/VHeader/VHeaderInternal.vue [hunk 2]
```diff
@@ -148,6 +151,13 @@ export default defineComponent({
       deactivateFocusTrap,
     })
 
+    const eventedOnTriggerClick = (...args) => {
```
Comment (obulat) [inline frontend/src/components/VHeader/VHeaderInternal.vue:154]: `onTriggerClick` does not take any arguments. I don't think we need `...args` here.

**Review by obulat — CHANGES_REQUESTED** [review]
This change is working well.
The blocker here is the eslintignore change. And it would be really nice to add a unit test.

**Comment by zackkrida** [comment]
Drafting while I add a unit test.

**Comment by zackkrida** [comment]
@obulat and @dhruvkb this is ready for review. I've added a test spec for the internal header with a unit test for this event.

**Review by obulat — APPROVED** [review]
Perfect! Works well both on the homepage and on the content pages. And **more** tests 🎉

**Pull request merged**